### PR TITLE
XW-2310 | VE skip saving dialog for page creation

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/XW-2310

* Auto-save on saving modal for page creation
* Move license under editor from saving modal

Ping @Wikia/x-wing

### CODE REVIEW

File: extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js [hunk 1]
```diff
@@ -85,6 +86,28 @@ ve.init.wikia.ViewPageTarget.static.actionsToolbarConfig = [
...
+	if ( this.$license ) {
+		this.$license.remove();
+		this.$license = null;
+	}
+
+	// Parent method
+	return ve.init.mw.ViewPageTarget.prototype.tearDownSurface.call( this, noAnimate );
+};
+
 //ve.init.wikia.ViewPageTarget.static.surfaceCommands.push( 'wikiaSourceMode' );
```
Comment (WikiaTech) [inline extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js [hunk 2]
```diff
@@ -460,11 +455,13 @@ ve.ui.MWSaveDialog.prototype.initialize = function () {
...
+
 			// Old messages should not persist
 			this.clearAllMessages();
 			this.swapPanel( 'save' );
 			// Update save button label
-			this.actions.forEach( { actions: 'save' }, function ( action ) {
+			this.getActions().forEach( { actions: 'save' }, function ( action ) {
 				action.setLabel(
 					ve.msg(
 						// TODO: Actually populate this.resotring with information, right now it is
```
Comment (WikiaTech) [inline extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* CRITICAL 1 critical
* MAJOR 7 major
* INFO 3 info

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [ve.ui.MWSaveDialog.js#L194](https://github.com/Wikia/app/blob/c0230749d7c17ac65d1cf8286ebf9a03dcfbf245/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L194): End this switch case with an unconditional break, continue, return or throw statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANonEmptyCaseWithoutBreak)
1. MAJOR [ve.ui.MWSaveDialog.js#L20](https://github.com/Wikia/app/blob/c0230749d7c17ac65d1cf8286ebf9a03dcfbf245/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L20): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.MWSaveDialog.js#L155](https://github.com/Wikia/app/blob/c0230749d7c17ac65d1cf8286ebf9a03dcfbf245/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L155): Function has a complexity of 13 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [ve.ui.MWSaveDialog.js#L177](https://github.com/Wikia/app/blob/c0230749d7c17ac65d1cf8286ebf9a03dcfbf245/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L177): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [ve.ui.MWSaveDialog.js#L185](https://github.com/Wikia/app/blob/c0230749d7c17ac65d1cf8286ebf9a03dcfbf245/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L185): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. MAJOR [ve.ui.MWSaveDialog.js#L213](https://github.com/Wikia/app/blob/c0230749d7c17ac65d1cf8286ebf9a03dcfbf245/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L213): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. MAJOR [ve.init.wikia.ViewPageTarget.js#L18](https://github.com/Wikia/app/blob/c0230749d7c17ac65d1cf8286ebf9a03dcfbf245/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js#L18): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. INFO [ve.ui.MWSaveDialog.js#L377](https://github.com/Wikia/app/blob/c0230749d7c17ac65d1cf8286ebf9a03dcfbf245/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L377): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)
1. INFO [ve.init.wikia.ViewPageTarget.js#L358](https://github.com/Wikia/app/blob/c0230749d7c17ac65d1cf8286ebf9a03dcfbf245/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js#L358): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Review by hakubo — APPROVED** [review]
LGTM as a hack :)

File: extensions/VisualEditor/wikia/modules/ve/ui/styles/ve.ui.WikiaLicense.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+@import "skins/shared/color";
+@import "color";
+
+.ve-ui-wikia-license {
+	background-color: rgba($color-black, .1);
+	border-top: 1px solid $color-page-border;
+	color: $color-ve-text-secondary;
+	font-size: 0.85em;
```
Comment (Warkot) [inline extensions/VisualEditor/wikia/modules/ve/ui/styles/ve.ui.WikiaLicense.scss]: https://github.com/Wikia/guidelines/blob/master/SCSS/CodingConventions.md#leadingzero …but, as it's a dirty hack, I can live with a leading zero. ;)

**Review by Warkot — APPROVED** [review]

File: extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js (hunk 1 above)
Comment (WikiaTech) [inline extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js:111]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js (hunk 2 above)
Comment (WikiaTech) [inline extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js:467]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* CRITICAL 1 critical
* MAJOR 7 major
* INFO 3 info

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [ve.ui.MWSaveDialog.js#L194](https://github.com/Wikia/app/blob/788950b28c39aabdb3ffccf722b91a360651fdca/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L194): End this switch case with an unconditional break, continue, return or throw statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANonEmptyCaseWithoutBreak)
1. MAJOR [ve.ui.MWSaveDialog.js#L20](https://github.com/Wikia/app/blob/788950b28c39aabdb3ffccf722b91a360651fdca/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L20): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.MWSaveDialog.js#L155](https://github.com/Wikia/app/blob/788950b28c39aabdb3ffccf722b91a360651fdca/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L155): Function has a complexity of 13 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [ve.ui.MWSaveDialog.js#L177](https://github.com/Wikia/app/blob/788950b28c39aabdb3ffccf722b91a360651fdca/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L177): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [ve.ui.MWSaveDialog.js#L185](https://github.com/Wikia/app/blob/788950b28c39aabdb3ffccf722b91a360651fdca/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L185): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. MAJOR [ve.ui.MWSaveDialog.js#L213](https://github.com/Wikia/app/blob/788950b28c39aabdb3ffccf722b91a360651fdca/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L213): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. MAJOR [ve.init.wikia.ViewPageTarget.js#L18](https://github.com/Wikia/app/blob/788950b28c39aabdb3ffccf722b91a360651fdca/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js#L18): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. INFO [ve.ui.MWSaveDialog.js#L377](https://github.com/Wikia/app/blob/788950b28c39aabdb3ffccf722b91a360651fdca/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L377): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)
1. INFO [ve.init.wikia.ViewPageTarget.js#L358](https://github.com/Wikia/app/blob/788950b28c39aabdb3ffccf722b91a360651fdca/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js#L358): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
- CRITICAL 1 critical
- MAJOR 7 major
- INFO 3 info

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. CRITICAL [ve.ui.MWSaveDialog.js#L194](https://github.com/Wikia/app/blob/8c3333f9993fc3a4c48eea1b1a8b7a6873063b6c/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L194): End this switch case with an unconditional break, continue, return or throw statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANonEmptyCaseWithoutBreak)
2. MAJOR [ve.ui.MWSaveDialog.js#L20](https://github.com/Wikia/app/blob/8c3333f9993fc3a4c48eea1b1a8b7a6873063b6c/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L20): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
3. MAJOR [ve.ui.MWSaveDialog.js#L155](https://github.com/Wikia/app/blob/8c3333f9993fc3a4c48eea1b1a8b7a6873063b6c/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L155): Function has a complexity of 13 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
4. MAJOR [ve.ui.MWSaveDialog.js#L177](https://github.com/Wikia/app/blob/8c3333f9993fc3a4c48eea1b1a8b7a6873063b6c/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L177): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
5. MAJOR [ve.ui.MWSaveDialog.js#L185](https://github.com/Wikia/app/blob/8c3333f9993fc3a4c48eea1b1a8b7a6873063b6c/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L185): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
6. MAJOR [ve.ui.MWSaveDialog.js#L213](https://github.com/Wikia/app/blob/8c3333f9993fc3a4c48eea1b1a8b7a6873063b6c/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L213): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
7. MAJOR [ve.init.wikia.ViewPageTarget.js#L18](https://github.com/Wikia/app/blob/8c3333f9993fc3a4c48eea1b1a8b7a6873063b6c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js#L18): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
8. INFO [ve.ui.MWSaveDialog.js#L377](https://github.com/Wikia/app/blob/8c3333f9993fc3a4c48eea1b1a8b7a6873063b6c/extensions/VisualEditor/modules/ve-mw/ui/dialogs/ve.ui.MWSaveDialog.js#L377): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)
9. INFO [ve.init.wikia.ViewPageTarget.js#L358](https://github.com/Wikia/app/blob/8c3333f9993fc3a4c48eea1b1a8b7a6873063b6c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js#L358): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Pull request merged**